frontend/src/metabase/admin/tasks/components/Logs/Logs.tsx

```diff
@@ -66,66 +68,85 @@ const LogsBase = ({
+              placeholder={t`Filter logs`}
+              rightSection={
+                query.length > 0 ? (
+                  <Button
+                    aria-label={t`Clear`}
+                    c="text-dark"
+                    leftSection={<Icon name="close" />}
+                    size="xs"
+                    variant="subtle"
+                    onClick={() => {
+                      patchUrlState({ query: "" });
+                      refollow();
+                    }}
+                  />
+                ) : undefined
+              }
+              value={query}
+              w={220} // set width to prevent CLS when the "Clear" button appears/disappears
+              onChange={(event) => {
+                patchUrlState({ query: event.target.value });
                 refollow();
               }}
-            >
-              <Option value="ALL" key="ALL">{t`All Metabase processes`}</Option>
-              {processUUIDs.map((uuid) => (
-                <Option key={uuid} value={uuid}>
-                  <code>{uuid}</code>
-                </Option>
-              ))}
-            </Select>
-          )}
+            />
+
+            {processUUIDs.length > 1 && (
+              <Select
+                defaultValue="ALL"
+                value={process}
+                width={400}
+                onChange={(e: { target: { value: string } }) => {
+                  patchUrlState({ process: e.target.value });
+                  refollow();
+                }}
+              >
+                <Option
+                  value="ALL"
+                  key="ALL"
+                >{t`All Metabase processes`}</Option>
+                {processUUIDs.map((uuid) => (
+                  <Option key={uuid} value={uuid}>
+                    <code>{uuid}</code>
+                  </Option>
+                ))}
+              </Select>
+            )}
+          </Flex>
+
+          <Flex align="center" gap="md">
+            <Button
+              component={Link}
+              to="/admin/troubleshooting/logs/levels"
+              leftSection={<Icon name="pulse" />}
+              variant="default"
+            >{t`Customize log levels`}</Button>
+
+            <Button
+              disabled={logText.length === 0}
+              leftSection={<Icon name="download" />}
+              variant="filled"
+              onClick={handleDownload}
+            >{t`Download`}</Button>
+          </Flex>
         </Flex>
 
-        <Button
-          disabled={logText.length === 0}
-          leftSection={<Icon name="download" />}
-          variant="filled"
-          onClick={handleDownload}
-        >{t`Download`}</Button>
-      </Flex>
+        <LogsContent id="logs-content" ref={scrollRef} onScroll={onScroll}>
+          {displayLogs}
+        </LogsContent>
+      </LogsContainer>
 
-      <LogsContent id="logs-content" ref={scrollRef} onScroll={onScroll}>
-        {displayLogs}
-      </LogsContent>
-    </LogsContainer>
+      {
```

Comment:
I had to move this block out of `LogsContainer` (which is `LoadingAndErrorWrapper`) because it caused a never-ending loop in unit tests